core/src/commonMain/kotlin/Filter.kt

```diff
@@ -1,10 +1,125 @@
+     */
+    public data class Service(
+        public val uuid: Uuid,
+    ) : Filter()
+
+    /**
+     * | Platform   | Supported | Details                                   |
+     * |------------|:---------:|-------------------------------------------|
+     * | Android    | Yes       | Supported natively                        |
+     * | Apple      | Yes       | Support provided by Kable via flow filter |
+     * | JavaScript | Yes       | Supported natively                        |
+     *
+     * https://developer.android.com/reference/android/bluetooth/le/ScanFilter.Builder#setDeviceName(java.lang.String)
+     */
+    public data class Name(
+        public val name: String,
+    ) : Filter()
+
+    /**
+     * | Platform   | Supported | Details                                   |
+     * |------------|:---------:|-------------------------------------------|
+     * | Android    | Yes       | Support provided by Kable via flow filter |
+     * | Apple      | Yes       | Support provided by Kable via flow filter |
+     * | JavaScript | Yes       | Supported natively                        |
+     */
+    public data class NamePrefix(val prefix: String) : Filter()
+
+    /**
+     * | Platform   | Supported | Details                                |
+     * |------------|:---------:|----------------------------------------|
+     * | Android    | Yes       | Supported natively                     |
+     * | Apple      | No        | Throws [UnsupportedOperationException] |
+     * | JavaScript | No        | Throws [UnsupportedOperationException] |
+     *
+     * https://developer.android.com/reference/android/bluetooth/le/ScanFilter.Builder#setDeviceAddress(java.lang.String)
+     */
+    public data class Address(
+        public val address: String,
+    ) : Filter()
+
+    /**
+     * | Platform   | Supported | Details                                   |
+     * |------------|:---------:|-------------------------------------------|
+     * | Android    | Yes       | Supported natively                        |
+     * | Apple      | Yes       | Support provided by Kable via flow filter |
+     * | JavaScript | Yes       | Supported natively                        |
+     *
+     * https://developer.android.com/reference/android/bluetooth/le/ScanFilter.Builder#setManufacturerData(int,%20byte[],%20byte[])
+     */
+    public class ManufacturerData(
+        /** A negative [id] is considered an invalid id. */
+        public val id: Int,
+
+        public val data: ByteArray,
+
+        /**
+         * For any bit in the mask, set it to 1 if it needs to match the corresponding bit in [data], otherwise set it
+         * to 0. If [dataMask] is not `null`, then it must have the same length as [data].
+         */
+        public val dataMask: ByteArray?,
+    ) : Filter() {
+        init {
+            if (dataMask != null) checkDataAndMask(data, dataMask)
+        }
+    }
```

Comment:
Maybe the comment is throwing me off, but I got very confused that a `ByteArray` is a bit-mask?

I'm not familiar with all the use cases, but it makes me wonder if this Is too specialized and perhaps the raw lambda comparator should be exposed for this `ManufacturerData` filter API instead?

Reply:
The lamba comparator is a cool idea, but strangely the bytes as a mask is the paradigm used natively for both Android and JavaScript. 🤷 

I'm inclined to get the feature in as the byte mask for now (since it is likely familiar for advanced users who have used the feature natively on either Android or JS) — then maybe sometime in the future add additional helpers that provide the lambda approach?

Reply:
I can rewrite/update the comment if you think that would be helpful?

Reply:
@davidtaylor-juul I've updated the documentation, let me know if that clarifies things?

https://github.com/JuulLabs/kable/blob/3bd19ddab03b9025268989668b017d1652f376c5/core/src/commonMain/kotlin/Filter.kt#L72-L109

Reply:
> the bytes as a mask is the paradigm used natively for both Android

Was not aware of this, and I also mis-read the comparison logic on first pass :-/  Makes perfect sense now.